### PR TITLE
limit console cmd length

### PR DESCRIPTION
**Pull request by knightXun**
For Security, Limit the max length of a single cmd.

### CODE REVIEW

File: src/console/CliManager.cpp [hunk 1]
```diff
@@ -142,15 +144,28 @@ void CliManager::loop() {
...
+            if (lineSize > kMaxCmdLen) {
+                cmdTooLarge = true; 
+                break;
+            }
+            line.resize(lineSize);
             cmd += line;
+            if(cmd.length() > kMaxCmdLen) {
+                cmdTooLarge = true; 
+                break;
+            }
```
Comment (dutor) [inline src/console/CliManager.cpp]: Only catch the case there are line breaks?
I think you could just check right before the command gets executed/processed.

**Comment by dangleptr** [comment]
Any progress about this pr?  
@knightXun @dutor @laura-ding

**Comment by Judy1992** [comment]
we should take the length limitation into consideration as a whole instead of just limiting the length of cmd, e.g. r we gonna limit the length of string prop.  Maybe do in at parser 🤷‍♂️🤷‍♂️

**Comment by jude-zhu** [comment]
we should take it into consideration as a whole instead of just limiting the length of cmd. For example. r we gonna limit the length of string prop. Maybe do it at parser? 🤷‍♂️🤷‍♂️

**Pull request closed without merging**